### PR TITLE
Show append instructions on exercise page

### LINKED ISSUE #6063: Exercise Overview: Instructions are missing the append part
The exercise overview page for an unfinished exercise shows the instructions of the exercise, e.g. https://exercism.org/tracks/go/exercises/simple-cipher (if you have not done that exercise). 

Currently it only shows the content of instructions.md. The content of instructions.append.md is missing. This causes problems for students that read the instructions there before starting the exercise. They don't notice that there is additional content after starting in the online editor or downloading via CLI.

### PR DESCRIPTION
**Pull request by ErikSchierboom**
**Ready to merge**

---

Closes https://github.com/exercism/exercism/issues/6063

### CODE REVIEW

File: app/models/git/exercise.rb [hunk 1]
```diff
@@ -2,16 +2,16 @@ module Git
...
 
     delegate :head_sha, :lookup_commit, :head_commit, to: :repo
 
-    git_filepaths instructions: ".docs/instructions.md",
-      instructions_append: ".docs/instructions.append.md",
-      introduction: ".docs/introduction.md",
-      introduction_append: ".docs/introduction.append.md",
-      hints: ".docs/hints.md",
-      config: ".meta/config.json"
+    git_filepath :instructions, file: ".docs/instructions.md", append_file: ".docs/instructions.append.md"
```
Comment (ErikSchierboom) [inline app/models/git/exercise.rb:9]: I changed `git_filepaths` to `git_filepath` to be able to easily add configuration options

Comment (iHiD) [reply app/models/git/exercise.rb:9]: Nice. This is good.

File: app/models/git/has_git_filepath.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+      # Define a <field> method that stored a memoized version of the contents
+      # of the file with the specified filepath as retrieved from Git
+      define_method field do
+        iv = "@__#{field}__"
+        return instance_variable_get(iv) if instance_variable_defined?(iv)
+
+        file_content = repo.send(read_method, commit, absolute_filepath(file))
+        file_content << repo.send(read_method, commit, absolute_filepath(append_file)) if append_file
+        file_content.strip! unless json_file
```
Comment (iHiD) [inline app/models/git/has_git_filepath.rb]: Really nice. Do we need to add whitespace between them?

Comment (ErikSchierboom) [reply app/models/git/has_git_filepath.rb]: I don't remember what we decided. But we can, yes

Comment (iHiD) [reply app/models/git/has_git_filepath.rb]: Only because on the old rendering bit it looks like they have a line of whitespace between them.

Comment (ErikSchierboom) [reply app/models/git/has_git_filepath.rb]: I've now separated them with "\n\n".

File: app/commands/solution/generate_readme_file.rb [hunk 3]
```diff
@@ -10,7 +10,7 @@ def call
         introduction,
         instructions,
         source
-      ].compact_blank.join("\n\n")
+      ].compact_blank.join("\n\n").gsub(/\n{3,}/, "\n\n")
```
Comment (ErikSchierboom) [inline app/commands/solution/generate_readme_file.rb]: The `gsub` call is there due to removal of h1 headings leaving extra blank lines if they are followed by an empty line. Want me to remove empty lines following h1 header removals instead?

File: app/models/git/has_git_filepath.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
+module Git
+  module HasGitFilepath
+    # To be able to include this module, the including class must have:
+    # - A member named "repo" of type Git::Repository
+    # - A method named "absolute_filepath" that takes a relative path
+    #   to a git file and returns its absolute path
+    def git_filepath(field, file:, append_file: nil)
+      json_file = file.end_with?('.json')
+      json_file ? read_method = "read_json_blob" : read_method = "read_text_blob"
```
Comment (iHiD) [inline app/models/git/has_git_filepath.rb]:
```suggestion
      read_method = json_file ? "read_json_blob" : "read_text_blob"
```

File: app/commands/solution/generate_readme_file.rb (hunk 3 above)
Comment (ErikSchierboom) [reply app/commands/solution/generate_readme_file.rb]: I've fixed this downstream when preprocessing the markdown

**Pull request merged**